cli/pkg/kctrl/cmd/app/init/git.go

```diff
@@ -11,62 +11,55 @@ import (
 	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
 )
 
-type GitStep struct {
+type GitConfiguration struct {
 	ui           cmdcore.AuthoringUI
-	vendirConfig vendirconf.Config
+	vendirConfig *VendirConfig
 }
 
-func NewGitStep(ui cmdcore.AuthoringUI, vendirConfig vendirconf.Config) *GitStep {
-	return &GitStep{
-		ui:           ui,
-		vendirConfig: vendirConfig,
-	}
+func NewGitConfiguration(ui cmdcore.AuthoringUI, vendirConfig *VendirConfig) *GitConfiguration {
+	return &GitConfiguration{ui: ui, vendirConfig: vendirConfig}
 }
 
-func (g *GitStep) PreInteract() error { return nil }
-
-func (g *GitStep) Interact() error {
-	contents := g.vendirConfig.Directories[0].Contents
+func (g *GitConfiguration) Configure() error {
+	contents := g.vendirConfig.Contents()
 	if contents == nil {
-		err := g.initializeContentWithGit()
+		err := g.initializeContentWithGit(contents)
```

Comment:
As `g` already has the information about `contents`, IMO we can skip passing contents.

Reply:
Sticking to this because all of the subsequent calls use contents, and we would end up calling `Contents()` multiple times. Can switch it up if you feel stronglyabout this